server/util/fastcopy/fastcopy_windows_test.go

```diff
@@ -0,0 +1,162 @@
+		&flags,                    // lpFileSystemFlags
+		nil,                       // lpFileSystemNameBuffer
+		0,                         // nFileSystemNameSize
+	)
+	require.NoError(t, err, "GetVolumeInformationByHandleW should succeed")
+
+	return flags&FILE_SUPPORTS_BLOCK_REFCOUNTING != 0
+}
+
+func TestCloneWindows(t *testing.T) {
+	if runtime.GOOS != "windows" {
+		t.Skipf("test runs on windows only")
+	}
+	flags.Set(t, "executor.enable_fastcopy_reflinking", true)
+
+	// Try to find a DevDrive for optimal CoW testing
+	hasBlockCloning := isBlockCloningSupported(t)
+	if !hasBlockCloning {
+		t.Skipf("No DevDrive found, skipping test for block cloning support")
+	}
+
+	t.Logf("DevDrive found at %s - testing with block cloning support", DevDrivePath)
+	testDir, err := os.MkdirTemp(DevDrivePath, "test-clone-windows-*")
+	require.NoError(t, err, "Failed to create temporary directory for test")
+	source := testfs.MakeTempFile(t, testDir, "test-source-*.txt")
+	target := filepath.Join(testDir, "test_target.txt")
+
+	// Ensure target doesn't exist
+	_, err = os.Stat(target)
+	require.Error(t, err)
+	require.True(t, errors.Is(err, os.ErrNotExist))
+
+	// Perform the clone operation
+	err = fastcopy.Clone(source, target)
+	require.NoError(t, err, "Clone should succeed on DevDrive with ReFS")
+	t.Cleanup(func() { os.Remove(target) })
+
+	// Verify target file exists and has correct content
+	_, err = os.Stat(target)
+	require.NoError(t, err, "target file should exist after clone")
+
+	// Read original test data for comparison
+	testData, err := os.ReadFile(source)
+	require.NoError(t, err)
+
+	targetData, err := os.ReadFile(target)
+	require.NoError(t, err)
+	require.Equal(t, testData, targetData, "target file should have same content as source")
+
+	// Test CoW behavior: modify target file and ensure source is unchanged
+	f, err := os.OpenFile(target, os.O_RDWR, 0)
+	require.NoError(t, err)
+	defer f.Close()
+
+	modifiedData := []byte("Modified data to test CoW")
+	_, err = f.WriteAt(modifiedData, 0)
+	require.NoError(t, err)
+
+	// Verify source file is unchanged (CoW behavior)
+	sourceData, err := os.ReadFile(source)
+	require.NoError(t, err)
+	require.Equal(t, testData, sourceData, "source file should be unchanged after modifying target (CoW)")
+
+	// Verify target file has the modified data
+	targetDataAfterModify, err := os.ReadFile(target)
+	require.NoError(t, err)
+	require.NotEqual(t, testData, targetDataAfterModify, "target file should have modified data")
+}
+
+func TestCloneWindowsCopyFileW(t *testing.T) {
+	if runtime.GOOS != "windows" {
+		t.Skipf("test runs on windows only")
+	}
+	flags.Set(t, "executor.enable_fastcopy_reflinking", false)
+
+	ws := testfs.MakeTempDir(t)
+	source := testfs.MakeTempFile(t, ws, "test data for CopyFileW")
+	target := path.Join(ws, "copyfilew_target.txt")
+
+	// Ensure target doesn't exist
+	_, err := os.Stat(target)
+	require.Error(t, err)
+	require.True(t, errors.Is(err, os.ErrNotExist))
```

Comment:
This should use `require.ErrorIs` (in the other method as well)

Reply:
done